### PR TITLE
Add a Toggle to the Status Bars w/ a Delay to show only in Combat

### LINKED ISSUE #7436: Hiding Status Bars when out of combat!
Is your feature request related to a problem?:
Not quite, more so inconvenience. 

Describe the solution you'd like:
A toggle that would make the status bars disappear automatically out of combat after X amount of seconds!

Describe alternatives you've considered:
The only alternative would be to go manually toggle the entire plugin constantly, depending on whether I'm doing skilling or combat activities. 

Additional context:
Whilst this is a bit of a first world problem, I do feel these big bright bars are a bit intrusive when I'm just skilling. How intrusive they are will depend on interface settings, but when using the "Bottom Line" setting with the collapsed rocks, they do feel a bit in your face.

### PR DESCRIPTION
**Pull request by ermalsh**
This is to only show the Status Bars only when in Combat, and hides it based on the delay set.
Closes #7436
Toggle Demo: https://gyazo.com/ee277e5f0d4bfad598a7a1049d972a67
Delay Demo: https://gyazo.com/421e2a4a725725f1b7b5e62f96a6a8a3

**Note: I know my diffs are messed up - I have updated my .gitconfig to match the settings that deathbeam provided to me to fix this for any future commits.**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 1]
```diff
@@ -47,10 +59,58 @@
...
+		if (!config.toggleRestorationBars())
+		{
+			overlayManager.add(overlay);
+			return;
+		}
+		else
+		{
+			final Actor interacting = client.getLocalPlayer().getInteracting();
+			final boolean isNpc = interacting instanceof NPC;
+			final int COMBAT_TIMEOUT = config.hideStatusBarDelay();
```
Comment (Nava2) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]:
```suggestion
			final int combatTimeout = config.hideStatusBarDelay();
```

The value is not a constant value, thus it shouldn't be written as one.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 2]
```diff
@@ -47,10 +59,58 @@
...
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!config.toggleRestorationBars())
+		{
+			overlayManager.add(overlay);
+			return;
+		}
+		else
+		{
+			final Actor interacting = client.getLocalPlayer().getInteracting();
```
Comment (Nava2) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: Consider moving all of this branch into a separate method for code cleanliness. Having multiple layers of indentation makes it harder to read whats happening.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 3]
```diff
@@ -60,4 +63,26 @@ default boolean enableRestorationBars()
 	{
 		return true;
 	}
+
+	@ConfigItem(
+		position = 4,
+		keyName = "toggleRestorationBars",
+		name = "Toggle to Hide when not in Combat",
```
Comment (Nava2) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: Consider:
```suggestion
		name = "Toggle to hide when not in combat",
```

**Comment by SirBenvil** [comment]
Is this still coming or was it cancelled? Forgive my curiosity :D

**Comment by SirBenvil** [comment]
Is this still coming or was it cancelled? Forgive my curiosity :D

**Comment by Malfuryent** [comment]
This will probably have to be adjusted to the new statusbars i'm currently proposing, see #7730 it's a rework of the current statusbars with added functionality.

**Comment by SirBenvil** [comment]
Is this ever making it to the live build? Its been almost a year and I still really want this feature, those bars still feel a bit in my face when using the interface "bottom line stones" and just skilling, etc.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 4]
```diff
@@ -32,6 +32,7 @@
 public interface StatusBarsConfig extends Config
 {
 	@ConfigItem(
+		position = 1,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: Remove the position parameters from the naturally-sorted configurations. They will be placed before configuration items with positive position values.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 5]
```diff
@@ -60,4 +63,26 @@ default boolean enableRestorationBars()
...
+		description = "Visually hides the Status Bars when player is out of combat."
+	)
+	default boolean toggleRestorationBars()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+		position = 5,
+		keyName = "hideStatusBarDelay",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: This is not a descriptive key name. It should reference that this is specifically the combat hiding delay.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java [hunk 6]
```diff
@@ -60,4 +63,26 @@ default boolean enableRestorationBars()
...
+		name = "Delay (seconds)",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: Both of the added configs could be condensed into one `int` config item which does not hide the bars if its value is 0.

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsConfig.java]: Also, now that `Units` have been added for config items, this should be annotated with the `Seconds` unit.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 7]
```diff
@@ -47,10 +59,63 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	private Instant lastCombatAction;
+
 	@Override
 	protected void startUp() throws Exception
 	{
-		overlayManager.add(overlay);
+	}
+
+	void updateLastCombatAction()
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: This is only called once and its function is clear without the function name's added context. `lastCombatAction = Instant.now();` should be moved to the single call site for this method.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 8]
```diff
@@ -47,10 +59,63 @@
...
+		}
+	}
+
+	private void hideStatusBar()
+	{
+		final Actor interacting = client.getLocalPlayer().getInteracting();
+		final boolean isNpc = interacting instanceof NPC;
+		final int combatTimeout = config.hideStatusBarDelay();
+
+		if (isNpc)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: `isNpc` is used exactly once, move its contents into this condition.

```suggestion
		if (interacting instanceof NPC)
```

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 9]
```diff
@@ -47,10 +59,63 @@
...
+			final List<String> npcMenuActions = Arrays.asList(npcComposition.getActions());
+			if (npcMenuActions.contains("Attack") && config.toggleRestorationBars())
+			{
+				updateLastCombatAction();
+				overlayManager.add(overlay);
+			}
+		}
+		else if (lastCombatAction != null)
+		{
+			if (Duration.between(getLastCombatAction(), Instant.now()).getSeconds() > combatTimeout)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: `combatTimeout` is used exactly once, move its contents into this condition.

Also, the `else if (lastCombatAction != null)` block contains only this if statement, combine these conditions.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 10]
```diff
@@ -47,10 +59,63 @@
...
+			{
+				overlayManager.remove(overlay);
+			}
+		}
 	}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: All of this code should be added below `shutDown()` and `provideConfig()`.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 11]
```diff
@@ -47,10 +59,63 @@
...
+	private void hideStatusBar()
+	{
+		final Actor interacting = client.getLocalPlayer().getInteracting();
+		final boolean isNpc = interacting instanceof NPC;
+		final int combatTimeout = config.hideStatusBarDelay();
+
+		if (isNpc)
+		{
+			final NPC npc = (NPC) interacting;
+			final NPCComposition npcComposition = npc.getComposition();
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: These can be combined into `final NPCComposition comp = ((NPC) interacting).getComposition();`

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 12]
```diff
@@ -47,10 +59,63 @@
...
+		this.lastCombatAction = Instant.now();
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!config.toggleRestorationBars())
+		{
+			overlayManager.add(overlay);
+			return;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: This `return` does nothing.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 13]
```diff
@@ -47,10 +59,63 @@
...
+		}
+		else
+		{
+			hideStatusBar();
+		}
+	}
+
+	private void hideStatusBar()
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: This is not what this method does.

File: runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java [hunk 14]
```diff
@@ -47,10 +59,63 @@
...
+
+	void updateLastCombatAction()
+	{
+		this.lastCombatAction = Instant.now();
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!config.toggleRestorationBars())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java]: Don't invert this condition.

**Comment by SirBenvil** [comment]
To this day I still really want this option to be a reality, but if getting it to the main Runelite client is proving to be a challenge, could it perhaps be considered for the Plugin Hub?

**Pull request closed without merging**